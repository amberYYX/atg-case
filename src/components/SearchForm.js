import React, { useState } from 'react'

const SearchFrom = ({ handleSearchGameInfo }) => {
  const [gameType, setGameType] = useState('')

  const handleGameTypeChange = e => {
    e.preventDefault()
    setGameType(e.target.value.toUpperCase())
  }

  const searchGameInfo = e => {
    e.preventDefault()
    // console.log(`${gameType} in searchForm`)
    handleSearchGameInfo(gameType)
  }

  return (
    <>
      <form className="w-full max-w-sm m-auto" onSubmit={searchGameInfo}>
        <div className="flex items-center border-b border-blue-500 py-1">
          <input
            className="appearance-none text-2xl bg-transparent border-none w-full text-blue-100 mr-3 py-1 px-2 leading-tight focus:outline-none"
            type="text"
            placeholder="Input game type here"
            value={gameType}
            onChange={handleGameTypeChange}
          />
          <button
            className="flex-shrink-0 bg-blue-500 hover:bg-blue-700 border-blue-500 hover:border-blue-700 focus:outline-none text-xl border-4 text-white py-1 px-2 rounded"
            type="submit"
          >
            search
          </button>
        </div>
      </form>
    </>
  )
}

export default SearchFrom
