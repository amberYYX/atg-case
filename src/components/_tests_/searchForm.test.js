import React from 'react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import { fireEvent, render } from '@testing-library/react'
import SearchForm from '../SearchForm'

describe('SearchForm', () => {
  it('renders default correctly', () => {
    const handleSearchGameInfo = jest.fn()
    const tree = renderer.create(
      <SearchForm handleSearchGameInfo={handleSearchGameInfo} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('has the correct text with input', () => {
    const handleSearchGameInfo = jest.fn()
    const utils = render(
      <SearchForm handleSearchGameInfo={handleSearchGameInfo} />
    )

    const gameType = utils.getByPlaceholderText('Input game type here')
    const form = utils.container.querySelector('form')

    fireEvent.change(gameType, { target: { value: 'v4' } })
    fireEvent.submit(form)
    // fireEvent.click(submit)
    expect(handleSearchGameInfo.mock.calls).toHaveLength(1)
    expect(gameType.value).toBe('V4')
  })
})
