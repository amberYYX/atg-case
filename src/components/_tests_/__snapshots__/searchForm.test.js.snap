// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders default correctly 1`] = `
<form
  className="w-full max-w-sm m-auto"
  onSubmit={[Function]}
>
  <div
    className="flex items-center border-b border-blue-500 py-1"
  >
    <input
      className="appearance-none text-2xl bg-transparent border-none w-full text-blue-100 mr-3 py-1 px-2 leading-tight focus:outline-none"
      onChange={[Function]}
      placeholder="Input game type here"
      type="text"
      value=""
    />
    <button
      className="flex-shrink-0 bg-blue-500 hover:bg-blue-700 border-blue-500 hover:border-blue-700 focus:outline-none text-xl border-4 text-white py-1 px-2 rounded"
      type="submit"
    >
      search
    </button>
  </div>
</form>
`;
